packages/salesforcedx-vscode-apex/src/commands/forceApexTestRunCodeAction.ts

```diff
@@ -86,7 +91,50 @@ export class ApexLibraryTestRunExecutor extends LibraryCommandletExecutor<{
     );
     const humanOutput = new HumanReporter().format(result, this.codeCoverage);
     channelService.appendLine(humanOutput);
-    return true;
+
+    await this.handleDiagnostics(result);
+    return result.summary.outcome === 'Passed';
+  }
+
+  private async handleDiagnostics(result: TestResult) {
+    ApexLibraryTestRunExecutor.diagnostics.clear();
+    const projectPath = getRootWorkspacePath();
+    const project = await SfdxProject.resolve(projectPath);
+    const defaultPackage = project.getDefaultPackage().fullPath;
+
+    result.tests.forEach(test => {
+      if (test.diagnostic) {
+        const diagnostic = test.diagnostic;
+        const filepath = join(
+          defaultPackage,
+          'main',
+          'default',
+          'classes',
+          `${diagnostic.className}.cls` ?? ''
```

Comment:
It seems less than ideal to be hardcoding the path to the Apex test class `main/default/classes` - any ideas on how else we can get this information?

Reply:
Yeah this won't work because the class might not be in the default directory or even in main/default/classes. Unless we indexed the project ahead of time, the only option is to resolve the components of the project and find it. Resolution with the library is decently fast.

I have work in progress to let you do this:

```typescript
const class = resolveSource(packageDirectories).find(c => c.fullName === fullName && c.type === type);
```

but for now the API is kind of bad, so try this:

```typescript
let testClass: SourceComponent;
for (const packageDir of packageDirectories) {
  const components = ComponentSet.fromSource(packageDir);
  testClass = components.getSourceComponents({ fullName: 'TestClass', type: 'ApexClass' }).next().value
  if (testClass) {
    break;
  }
}
```

Reply:
Exactly, I wanted to handle the scenarios where the project structure is different than the default main/default/classes. This solution worked for me, thank you! :)